app/routes.py

```diff
@@ -1,2 +1,315 @@
+        try:
+            # flask puts the datetime in rfc 1123 format and I had to use parser.parse
+            dateutil.parser.parse(request_body["completed_at"])
+        except dateutil.parser.ParserError:
+            return True
+    return False
+
+
+@tasks_bp.route("/<int:task_id>", methods=["GET"])
+def get_task_by_id(task_id):
+    """
+    Returns a response with the task with given id as body and a 200 code
+    when the task is found
+    Returns a 404 with no response body in case the task is not found
+    """
+    task = Task.query.get_or_404(task_id)
+    return jsonify(task=task.as_dict())
+
+
+@tasks_bp.route("/<int:task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    """
+    Deletes the task with a given id and returns a 200 response with a success 
+    message
+    Returns a 404 with no response body in case the task is not found
+    """
+    task = Task.query.get_or_404(task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+    return make_response(jsonify(details=f"Task {task.task_id} \"{task.title}\" successfully deleted"), 200)
+
+
+@tasks_bp.route("/<int:task_id>", methods=["PUT"])
+def update_task(task_id):
+    """
+    Performs the update action and returns a 200 response with the newly updated task
+
+    If any of the expected field is missing in the post request body, it returns
+    a 400 response indicating invalid data
+
+    Returns a 404 with no response body in case the task is not found
+    """
+    task = Task.query.get_or_404(task_id)
+
+    request_body = request.get_json()
+
+    if invalid_post_request_body(request_body):
+        return make_response({"details": "Invalid data"}, 400)
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    task.completed_at = request_body["completed_at"]
+    db.session.commit()
+    return make_response(jsonify(task=task.as_dict()), 200)
+
+
+@tasks_bp.route("/<int:task_id>/mark_complete", methods=["PATCH"])
+def mark_task_complete(task_id):
+    """
+    Changes the task.completed_at value to todays date and returns a 200 response
+    with the updated task and calls the send_task_notification funtion
+    Returns a 404 with no response body in case the task is not found
+    """
+    task = Task.query.get_or_404(task_id)
+
+    task.completed_at = datetime.now(timezone.utc)
+    db.session.commit()
+
+    send_slack_task_notification(task)
+    return make_response(jsonify(task=task.as_dict()), 200)
+
+
+def send_slack_task_notification(task):
+    """
+    Sends a request to a slack bot to post the
+    "Someone just completed the task <TASK_TITLE>" to the task-notifications channel
+    in the configured slack workspace
+    """
+    SLACK_BOT_TOKEN = os.environ.get("SLACK_BOT_TOKEN")
+    text = f"Someone just completed the task {task.title}"
+    url = f"https://slack.com/api/chat.postMessage?channel=task-notifications&text={text}"
+
+    payload = ""
```

Comment:
payload is a common naming convention for a collection of data being sent thru an API like this. glad you used it here!